aws-cpp-sdk-core/source/client/AWSClient.cpp

```diff
@@ -304,6 +299,12 @@ HttpResponseOutcome AWSClient::AttemptOneRequest(const Aws::Http::URI& uri,
     std::shared_ptr<HttpResponse> httpResponse(
         m_httpClient->MakeRequest(*httpRequest, m_readRateLimiter.get(), m_writeRateLimiter.get()));
 
+    if(request.GetBody())
+    {
+        request.GetBody()->clear();
+        request.GetBody()->seekg(0);
+    }
+
```

Comment:
Hey, this change needs to happen in the success case only.
So this needs to be added as an else block like:

```cpp
    if (DoesResponseGenerateError(httpResponse))
    {
        AWS_LOGSTREAM_DEBUG(AWS_CLIENT_LOG_TAG, "Request returned error. Attempting to generate appropriate error codes from response");
        return HttpResponseOutcome(BuildAWSError(httpResponse));
    }
    else if(request.GetBody())
    {
        request.GetBody()->clear();
        request.GetBody()->seekg(0);
    }
```

Reply:
In case the retry parameter is set as true, then on failure, it'll result in more failures

Reply:
We should not clear the flags when the request fails. That's a dangerous assumption.